rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <VX/vx.h>
+#include <VX/vx_types.h>
+
+#include <array>
+#include <cstring>
+#include <memory>
+#include <queue>
+#include <vector>
+#if ENABLE_HIP
+#include "device_manager_hip.h"
+#include "hip/hip_runtime.h"
+#else
+#include "device_manager.h"
+#endif
+#include "commons.h"
+
+/*! \brief Converts Rocal Memory type to OpenVX memory type
+ *
+ * @param mem input Rocal type
+ * @return the OpenVX type associated with input argument
+ */
+vx_enum vx_mem_type(RocalMemType mem);
+
+/*! \brief Returns the size of the data type
+ *
+ * @param RocalTensorDataType input data type
+ * @return the OpenVX data type size associated with input argument
+ */
+vx_size tensor_data_size(RocalTensorDataType data_type);
+
+/*! \brief Holds the information about a rocalTensor */
+class rocalTensorInfo {
+public:
+    friend class rocalTensor;
+    enum class Type {
+        UNKNOWN = -1,
+        REGULAR = 0,
+        VIRTUAL = 1,
+        HANDLE = 2
+    };
+
+    // Default constructor
+    /*! initializes memory type to host and dimension as empty vector*/
+    rocalTensorInfo();
+
+    //! Initializer constructor with only fields common to all types (Image/ Video / Audio)
+    rocalTensorInfo(std::vector<size_t> dims, RocalMemType mem_type,
+                    RocalTensorDataType data_type);
+
+    // Setting properties required for Image / Video
+    void set_roi_type(RocalROIType roi_type) { _roi_type = roi_type; }
+    void set_data_type(RocalTensorDataType data_type) {
+        _data_type = data_type;
+        _data_size = (_data_size / _data_type_size);
+        _data_size *= data_type_size();
+    }
+    void set_max_dims() {
+        if (_layout != RocalTensorlayout::NONE) {
+            _max_dims.resize(2);  // Since 2 values will be stored in the vector
```

Comment:
max_dims is confusing. Probably better to rename to what it represent  (width, height?)

Reply:
The max dims will store the width and height in case of image tensors and frames or channels when it comes to Audio tensors hence we have retained the name max_dims

Reply:
In that case it might be better too rename it as _shape. There can be a tensor with dims (256, 3, 4, 4) where max_dims leads to confusing interpretation

Reply:
@rrawther I have renamed it as max_shape